kernel: bump 5.15 to 5.15.40

graysky2 opened the pull request:
All patches automatically rebased.

Build system: x86_64
Build-tested: bcm2711/RPi4B, mt7622/RT3200
Run-tested: bcm2711/RPi4B, mt7622/RT3200

Signed-off-by: John Audia <therealgraysky@proton.me>

graysky2 committed:
kernel: bump 5.15 to 5.15.39

Removed upstreamed:
  backport-5.15/850-v5.17-0001-PCI-pci-bridge-emul-Add-description-for-class_revisi.patch
  backport-5.15/850-v5.17-0002-PCI-pci-bridge-emul-Add-definitions-for-missing-capa.patch
  backport-5.15/850-v5.17-0003-PCI-aardvark-Add-support-for-DEVCAP2-DEVCTL2-LNKCAP2.patch
  backport-5.15/850-v5.17-0004-PCI-aardvark-Clear-all-MSIs-at-setup.patch
  backport-5.15/850-v5.17-0005-PCI-aardvark-Comment-actions-in-driver-remove-method.patch
  backport-5.15/850-v5.17-0006-PCI-aardvark-Disable-bus-mastering-when-unbinding-dr.patch
  backport-5.15/850-v5.17-0007-PCI-aardvark-Mask-all-interrupts-when-unbinding-driv.patch
  backport-5.15/850-v5.17-0008-PCI-aardvark-Fix-memory-leak-in-driver-unbind.patch
  backport-5.15/850-v5.17-0009-PCI-aardvark-Assert-PERST-when-unbinding-driver.patch
  backport-5.15/850-v5.17-0010-PCI-aardvark-Disable-link-training-when-unbinding-dr.patch
  backport-5.15/850-v5.17-0011-PCI-aardvark-Disable-common-PHY-when-unbinding-drive.patch
  pending-5.15/850-0001-PCI-aardvark-Replace-custom-PCIE_CORE_INT_-macros-wi.patch
  pending-5.15/850-0004-PCI-aardvark-Rewrite-IRQ-code-to-chained-IRQ-handler.patch
  pending-5.15/850-0005-PCI-aardvark-Check-return-value-of-generic_handle_do.patch
  pending-5.15/850-0006-PCI-aardvark-Make-MSI-irq_chip-structures-static-dri.patch
  pending-5.15/850-0007-PCI-aardvark-Make-msi_domain_info-structure-a-static.patch
  pending-5.15/850-0008-PCI-aardvark-Use-dev_fwnode-instead-of-of_node_to_fw.patch
  pending-5.15/850-0009-PCI-aardvark-Refactor-unmasking-summary-MSI-interrup.patch
  pending-5.15/850-0010-PCI-aardvark-Add-support-for-masking-MSI-interrupts.patch
  pending-5.15/850-0011-PCI-aardvark-Fix-setting-MSI-address.patc
  pending-5.15/850-0012-PCI-aardvark-Enable-MSI-X-support.patch
  pending-5.15/850-0013-PCI-aardvark-Add-support-for-ERR-interrupt-on-emulat.patch
  pending-5.15/850-0015-PCI-aardvark-Optimize-writing-PCI_EXP_RTCTL_PMEIE-an.patch
  pending-5.15/850-0016-PCI-aardvark-Add-support-for-PME-interrupts.patch
  pending-5.15/850-0017-PCI-aardvark-Fix-support-for-PME-requester-on-emulat.patch
  pending-5.15/850-0018-PCI-aardvark-Use-separate-INTA-interrupt-for-emulate.patch
  pending-5.15/850-0019-PCI-aardvark-Remove-irq_mask_ack-callback-for-INTx-i.patch
  pending-5.15/850-0020-PCI-aardvark-Don-t-mask-irq-when-mapping.patch
  pending-5.15/850-0021-PCI-aardvark-Drop-__maybe_unused-from-advk_pcie_disa.patch
  pending-5.15/850-0022-PCI-aardvark-Update-comment-about-link-going-down-af.patch

All patches automatically rebased.

Build system: x86_64
Build-tested: bcm2711/RPi4B, mt7622/RT3200
Run-tested: bcm2711/RPi4B, mt7622/RT3200

Signed-off-by: John Audia <therealgraysky@proton.me>

Boos4721 commented:
**WTF 5.150**

hauke commented on target/linux/generic/pending-5.15/850-0016-PCI-aardvark-Add-support-for-PME-interrupts.patch:
```diff
@@ -25,9 +25,9 @@ Signed-off-by: Marek Behún <kabel@kernel.org>
 
 --- a/drivers/pci/controller/pci-aardvark.c
 +++ b/drivers/pci/controller/pci-aardvark.c
-@@ -1480,6 +1480,18 @@ static void advk_pcie_handle_int(struct
```
Please remove this patch it is also upstream, see https://github.com/gregkh/linux/commit/cf7ef0e3029e43f85b525dd49420d56792a4e61f

hauke replied:
This is not resolved

graysky2 replied:
Should be now

hauke commented on target/linux/generic/backport-5.15/850-v5.17-0007-PCI-aardvark-Mask-all-interrupts-when-unbinding-driv.patch:
```diff
@@ -18,9 +18,9 @@ Signed-off-by: Lorenzo Pieralisi <lorenzo.pieralisi@arm.com>
+ 	/* Remove IRQ handler */
+ 	irq_set_chained_handler_and_data(pcie->irq, NULL, NULL);
  
 +	/* Disable MSI */
```
Please remove this patch, it was applied upstream, see https://github.com/gregkh/linux/commit/55e38c42051dc27224c5d27a5d81a8b85c324d0a

hauke commented on target/linux/generic/backport-5.15/850-v5.17-0011-PCI-aardvark-Disable-common-PHY-when-unbinding-drive.patch:
```diff
@@ -18,9 +18,9 @@ Signed-off-by: Lorenzo Pieralisi <lorenzo.pieralisi@arm.com>
-@@ -1734,6 +1734,9 @@ static int advk_pcie_remove(struct platf
- 	for (i = 0; i < OB_WIN_COUNT; i++)
- 		advk_pcie_disable_ob_win(pcie, i);
+@@ -1640,6 +1640,9 @@ static int advk_pcie_enable_phy(struct a
```
Please remove this patch, it was applied upstream, see https://github.com/gregkh/linux/commit/ac76c36aa417014b4e46d4d1dee7b15c39ccbf8c

hauke commented on target/linux/generic/backport-5.15/850-v5.17-0009-PCI-aardvark-Assert-PERST-when-unbinding-driver.patch:
```diff
@@ -20,7 +20,7 @@ Signed-off-by: Lorenzo Pieralisi <lorenzo.pieralisi@arm.com>
 --- a/drivers/pci/controller/pci-aardvark.c
 +++ b/drivers/pci/controller/pci-aardvark.c
-@@ -1721,6 +1721,10 @@ static int advk_pcie_remove(struct platf
+@@ -1919,6 +1919,10 @@ static int advk_pcie_remove(struct platf
```
Please remove this patch, it was already applied upstream, see https://github.com/gregkh/linux/commit/246a36b6637740f1be9c3cdd91f022b24b7402e7

hauke commented on target/linux/generic/backport-5.15/850-v5.17-0010-PCI-aardvark-Disable-link-training-when-unbinding-dr.patch:
```diff
@@ -20,7 +20,7 @@ Signed-off-by: Lorenzo Pieralisi <lorenzo.pieralisi@arm.com>
 
 --- a/drivers/pci/controller/pci-aardvark.c
 +++ b/drivers/pci/controller/pci-aardvark.c
-@@ -1725,6 +1725,11 @@ static int advk_pcie_remove(struct platf
```
Please remove this patch, it was already applied upstream, see https://github.com/gregkh/linux/commit/a346d08ca45e8475975187f5bab8540cb9ac75d2

hauke commented on target/linux/generic/backport-5.15/850-v5.17-0008-PCI-aardvark-Fix-memory-leak-in-driver-unbind.patch:
```diff
@@ -21,9 +21,9 @@ Signed-off-by: Lorenzo Pieralisi <lorenzo.pieralisi@arm.com>
 
 --- a/drivers/pci/controller/pci-aardvark.c
 +++ b/drivers/pci/controller/pci-aardvark.c
-@@ -1718,6 +1718,9 @@ static int advk_pcie_remove(struct platf
```
Please remove this patch, it was already applied upstream, see https://github.com/gregkh/linux/commit/af27f54e62243a6609c8e21c98b4e7eed4cde992

graysky2 commented:
@hauke - Thanks for the review.  I made the changes you called out.

hauke commented on target/linux/generic/backport-5.15/850-v5.17-0002-PCI-pci-bridge-emul-Add-definitions-for-missing-capa.patch:
```diff
@@ -21,9 +21,9 @@ Signed-off-by: Lorenzo Pieralisi <lorenzo.pieralisi@arm.com>
-@@ -270,6 +270,49 @@ struct pci_bridge_reg_behavior pcie_cap_
- 		.ro = GENMASK(15, 0) | PCI_EXP_RTSTA_PENDING,
- 		.w1c = PCI_EXP_RTSTA_PME,
+@@ -313,6 +313,49 @@ struct pci_bridge_reg_behavior pcie_cap_
```
Please remove this patch, it was applied upstream, see https://github.com/gregkh/linux/commit/739c052b9b26bc5abff3f6a5139292c00bc1964e

graysky2 commented:
Done.  I am surprised that the update script missed these.

graysky2 commented:
I only see two patches now and cannot find either of them in the [5.15.39 stable tree](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/log/?h=v5.15.39):
```
% find . -name '*aardv*.patch'|grep 5.15
./target/linux/generic/pending-5.15/851-0006-Revert-PCI-aardvark-Fix-initialization-with-old-Marv.patch
./target/linux/generic/pending-5.15/850-0023-PCI-aardvark-Make-main-irq_chip-structure-a-static-d.patch
```

hauke commented:
I did a diff between the new and the old content of the drivers/pci/controller/pci-aardvark.c file. I checked why it was different and found the problems.

Normally when I review these pull requests I closely look at the patches which change other that just changing some offsets.

1Jeff1 commented:
> I only see two patches now and cannot find either of them in the [5.15.39 stable tree](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/log/?h=v5.15.39):


@graysky2  850-0023-PCI-aardvark-Make-main-irq_chip-structure-a-static-d.patch https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.39&id=d951cbef2e4607868cc94b86ae6def03ec2fff4a

hauke commented:
850-0023-PCI-aardvark-Make-main-irq_chip-structure-a-static-d.patch is not included in kernel 5.15.39. 

The commit https://github.com/gregkh/linux/commit/d951cbef2e4607868cc94b86ae6def03ec2fff4a matches this patch: 
target/linux/generic/pending-5.15/850-0006-PCI-aardvark-Make-MSI-irq_chip-structures-static-dri.patch

graysky2 committed:
kernel: bump 5.15 to 5.15.40

All patches automatically rebased.

Build system: x86_64
Build-tested: bcm2711/RPi4B, mt7622/RT3200
Run-tested: bcm2711/RPi4B, mt7622/RT3200

Signed-off-by: John Audia <therealgraysky@proton.me>

hauke commented:
Thank you for your patches, I applied them to master in 6cdf974a3d0cfebf4a37aaf2876b2ece0f11363d and e156f1f862ff0c834df3b92a180598235e578e84.
